Core/Python/add_device_to_static_group.py

```diff
@@ -20,33 +20,158 @@
-        --password <pwd> --groupname "Random Test Group" --devicename "cmc1"
+   python add_device_to_static_group.py --ip <xx> --user <username>
+        --password <pwd> --groupname "Random Test Group" --devicenames "cmc1,host3,192.168.1.5"
 """
 
 import json
-import sys
 import argparse
 from argparse import RawTextHelpFormatter
 import urllib3
 import requests
 
 
-def add_device_to_static_group(ome_ip_address, ome_username, ome_password, group_name):
-    """ Authenticate with OME and enumerate groups """
+def get_device_list(ome_ip_address: str, headers: dict) -> list:
+    """
+    Retrieves a list of all devices being handled by this OME server
+
+    Args:
+        ome_ip_address: The IP address of the OME server
+        headers: Headers used for authentication to the OME server
+
+    Returns: A list of all devices managed by the this OME server
+
+    """
+
+    print("Retrieving a list of all devices...")
+    next_link_url = "https://%s/api/DeviceService/Devices" % ome_ip_address
+    device_data = None
+
+    while next_link_url is not None:
+        device_response = requests.get(next_link_url, headers=headers, verify=False)
+        next_link_url = None
+        if device_response.status_code == 200:
+            data = device_response.json()
+            if data['@odata.count'] <= 0:
+                print("No devices are managed by OME server: " + ome_ip_address + ". Exiting.")
+                exit(1)
+            if '@odata.nextLink' in data:
+                next_link_url = "https://%s" + data['@odata.nextLink']
+            if device_data is None:
+                device_data = data["value"]
+            else:
+                device_data += data["value"]
+        else:
+            print("Unable to retrieve device list from %s" % ome_ip_address)
+            exit(1)
+
+    return device_data
+
+
+def get_group_id_by_name(ome_ip_address: str, group_name: str, headers: dict) -> int:
+    """
+    Retrieves the ID of a group given its name.
+
+    Args:
+        ome_ip_address: The IP address of the OME server
+        group_name: The name of the group whose ID you want to resolve.
+        headers: Headers used for authentication to the OME server
+
+    Returns: Returns the ID of the group as an integer.
+
+    """
+
+    print("Searching for the requested group.")
+    groups_url = "https://%s/api/GroupService/Groups?$filter=Name eq '%s'" % (ome_ip_address, group_name)
+
+    group_response = requests.get(groups_url, headers=headers, verify=False)
+
+    if group_response.status_code == 200:
+        json_data = json.loads(group_response.content)
+
+        if json_data['@odata.count'] > 1:
+            print("WARNING: We found more than one name that matched the group name: " + group_name +
+                  ". We are picking the first entry.")
+        if json_data['@odata.count'] == 1 or json_data['@odata.count'] > 1:
+            group_id = json_data['value'][0]['Id']
+            if not isinstance(group_id, int):
```

Comment:
Are you seeing errors where integer values aren't returned for valid groups? What version of OME are you using?

Reply:
No, but I'm an overly cautious programmer 😂

I'm a fan of strict typing 🤷‍♂️. Maybe it's because I started on C back in the day hahaha